### PR TITLE
Filter tokens

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Fix some places where tokens weren't getting filtered out of Honeycomb payloads.

### Motivation and Context

Task: https://app.asana.com/0/1174274412967132/1200281743086565/f

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/filter_logging.rb [hunk 1]
```diff
@@ -146,8 +152,8 @@ def self.filter_honeycomb_data(fields)
 
     # RestClient
     if fields['name'].start_with?('restclient')
-      if fields.has_key?('restclient.headers') && fields['restclient.headers']['Authorization']
-        fields['restclient.headers']['Authorization'] = FILTERED
+      if fields.has_key?('restclient.request.header') && fields['restclient.request.header']['Authorization']
+        fields['restclient.request.header']['Authorization'] = FILTERED
```
Comment (sadleb) [inline lib/filter_logging.rb:156]: Did something change or was this never working?

File: lib/filter_logging.rb [hunk 2]
```diff
@@ -169,10 +175,17 @@ def self.filter_honeycomb_data(fields)
...
+    if fields.has_key?('aws.params.body')
       fields.delete('aws.params.body')
     end
 
+    # Don't send email bodies from ActionMailer.
+    # These often include sensitive information like tokens, and are hard to reliably
+    # filter, since they're MIME-encoded.
+    if fields.has_key?('deliver.action_mailer.mail')
+      fields.delete('deliver.action_mailer.mail')
+    end
```
Comment (sadleb) [inline lib/filter_logging.rb:187]: 👍

**Review by sadleb — APPROVED** [review]

**Review by sadleb — COMMENTED** [review]
Worth adding to `spec/lib/filter_logging.rb` ?

File: lib/filter_logging.rb (hunk 1 above)
Comment (rshipp) [reply lib/filter_logging.rb:156]: I changed it a while ago with the rest of the honeycomb fields.

**Comment by rshipp** [comment]
That spec isn't giving me any confidence right now, since I just found and fixed several issues that it didn't catch. I don't think it's worth maintaining.

**Pull request merged**